library/system/test/shared_lib_info_test.rb

```diff
@@ -0,0 +1,112 @@
+#! /usr/bin/env rspec
+
+require_relative "./test_helper"
+require "yast2/shared_lib_info"
+
+PROC_MAPS_PATH = File.join(__dir__, "data/proc-maps")
+
+def stored_proc_maps(scenario)
+  File.join(PROC_MAPS_PATH, "proc-maps-#{scenario}")
+end
+
+describe Yast::SharedLibInfo do
+  describe "#new" do
+    let(:subject) { described_class.new(maps_file) }
+
+    context "empty" do
+      let(:maps_file) { nil }
+
+      it "does not crash and burn" do
+        expect(subject.shared_libs).to eq []
+      end
+    end
+
+    context "with the Qt UI" do
+      let(:maps_file) { stored_proc_maps("qt") }
+
+      it "finds a lot of shared libs" do
+        expect(subject.shared_libs.size).to eq 117
```

Comment:
IMHO with "lot of shared libs" we could be less strict here, e.g. `> 100` would be enough.

(I try to not be that strict in tests for values which might change and do not need to be exact. It makes easier updating the testing data files.)

If we want to test that if finds all 117 shared libs in the file then we should change the description to something like `finds all loaded shared libs`.

Reply:
I was actually thinking about that; but if some bug in the parsing creeps in, the explicit check here will alert us. Weighing the pros and cons when I wrote that, the "but we are using a fixed file here that we have completely under our control" argument won. As long as that file doesn't change, the number also won't change; and when it does (which appears unlikely to me), it's easy to adapt that number in the test.

But you are right about the wording for the test description here; I'll change that.